internal/providers/openstack/openstack.go

```diff
@@ -91,6 +93,29 @@ func FetchConfig(f *resource.Fetcher) (types.Config, report.Report, error) {
 		return fetchConfigFromMetadataService(f)
 	})
 
+	// function literal to check both ctx.Done() as well as the amount
+	// of errors in the errChan to allow for non-blocking checks
+	watcher := func() bool {
+		select {
+		case <-ctx.Done():
+			if ctx.Err() == context.DeadlineExceeded {
+				f.Logger.Info("neither config drive nor metadata service were available in time. Continuing without a config...")
+			}
+			return true
+		default:
```

Comment:
Shouldn't this be something like `case <-errChan:`? Otherwise we'll be busy-looping checking the channel length. So then we'd increment some integer and compare that against `ndispatches` below.